ts/WoltLabSuite/Core/Ui/Search/Page.ts

```diff
@@ -106,4 +98,21 @@ export function init(objectType: string): void {
   // trigger click on init
   const link = dropdownMenu.querySelector('a[data-object-type="' + objectType + '"]') as HTMLAnchorElement;
   link.click();
+
+  searchInput.form!.addEventListener("submit", (event) => {
+    event.preventDefault();
+    submit(searchInput.form!, searchInput);
+  });
+}
+
+function submit(form: HTMLFormElement, input: HTMLInputElement): void {
+  const url = new URL(form.action);
+  url.search += url.search !== "" ? "&" : "?";
+  url.search += "q=" + encodeURIComponent(input.value.trim());
+
+  parameters.forEach((value, key) => {
+    url.search += "&" + encodeURIComponent(key) + "=" + encodeURIComponent(value);
+  });
```

Comment:
This could be much simpler as https://developer.mozilla.org/en-US/docs/Web/API/URLSearchParams.

Reply:
Unfortunately, using URLSearchParams leads to an undefined order of the parameters and to an unwanted encoding of the first parameter (`index.php?search/` -> `index.php?search%2F`).

Reply:
It should be fine if you just use it to append the additional parameters like this:

```
const url = new URL(form.action);
url.search += url.search !== "" ? "&" : "?";
url.search += (new URLSearchParams({q: input.value.trim(), ...parameters})).toString();
```